extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -389,20 +389,58 @@ export class AmpStoryPage extends AMP.BaseElement {
     if (autoAdvanceAttr !== null || storyNextUpParam === null) {
       return;
     }
+    addAttributesToElement(
+      this.element,
+      dict({
+        'auto-advance-after': storyNextUpParam,
+      })
+    );
+    this.listenAndUpdateAutoAdvanceDuration_();
+  }
+
+  /**
+   * If there's a video on the page, this sets a listener to update
+   * the TimeBasedAdvancement when the first video's duration becomes available.
+   * @private
+   */
+  listenAndUpdateAutoAdvanceDuration_() {
     const video = this.getFirstAmpVideo_();
     if (video === null) {
-      addAttributesToElement(
-        this.element,
-        dict({'auto-advance-after': storyNextUpParam})
-      );
       return;
     }
-    if (video.id === '') {
-      video.id = this.element.id + '_firstVideo';
+    whenUpgradedToCustomElement(video).then(() => {
+      video.getImpl().then((videoImpl) => {
```

Comment:
Small nit, we like to do promise chaining instead of nesting them to save of indentation levels (it also help with error catching in some scenarios but not here)

eg:

```
    whenUpgradedToCustomElement(video).then(() => {
      return video.getImpl();
    }).then((videoImpl) => {
      // ...
    });
```